aten/src/ATen/native/PointwiseOps.cpp

```diff
@@ -69,6 +69,13 @@ Tensor& addcdiv_out(
     const Tensor& tensor1,
     const Tensor& tensor2,
     Scalar value) {
+  if (isIntegralType(tensor1.scalar_type(), /*includeBool=*/ true)
+      && isIntegralType(tensor2.scalar_type(), /*includeBool=*/ true)) {
+    TORCH_WARN_ONCE(
+      "Integer division with addcdiv is deprecated. ",
+      "In a future release addcdiv will perform a true division of ",
+      "tensor1 and tensor2.");
```

Comment:
err...now that I think about it, does this even do type promotion with `value`?

It doesn't seem to on my version ('1.5.0.dev20200210')

```
>>> torch.addcdiv(torch.tensor([2,3]), torch.tensor([2,3]), torch.tensor([2,3], value=1.0)
tensor([3, 4])
```

I guess that's fine, and we sort of work around the issue because in the future we'll just mostly use the default dtype.

Reply:
So, I think it's fine to leave this as is, but suggest what they should use instead, ie.: `self + value * tensor1 // tensor2` (although this technically won't be the same because of type promotion).

Reply:
It will be the same where the current addcdiv is defined, though. Sounds like a good recommendation.